torch_geometric/loader/link_neighbor_loader.py

```diff
@@ -258,7 +305,12 @@ def __init__(
         edge_type, edge_label_index = get_edge_label_index(
             data, edge_label_index)
 
-        if neighbor_sampler is None:
+        # Corner Case: When data doesn't have any edges
+        # return a subset of nodes.
+        if not has_edge_index(data):
+            self.neighbor_sampler = LinkSampler(data, neg_sampling_ratio,
```

Comment:
```suggestion
            self.neighbor_sampler = LinkSampler(data, self.neg_sampling_ratio,
```
or modify it below, just for consistency